core/src/main/java/org/arquillian/smart/testing/scm/git/GitChangeResolver.java

```diff
@@ -59,23 +59,34 @@ public GitChangeResolver(File projectDir, Scm scm) {
     public GitChangeResolver(File dir, String previous, String head) {
         this.previous = previous;
         this.head = head;
-        final FileRepositoryBuilder builder = new FileRepositoryBuilder();
-        try {
-            this.git = new Git(builder.readEnvironment().findGitDir(dir).build());
-        } catch (IOException e) {
-            throw new IllegalArgumentException("Unable to find git repository for path " + dir.getAbsolutePath(), e);
+        final FileRepositoryBuilder fileRepositoryBuilder = getFileRepositoryBuilder(dir);
+        if (fileRepositoryBuilder.getGitDir() != null) {
+            try {
+                this.git = new Git(fileRepositoryBuilder.build());
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+            this.repoRoot = git.getRepository().getDirectory().getParentFile();
+        } else {
+            logger.warn(String.format("Unable to find a git repository for the path %s - new, changed, affected strategies won't work properly."
+                + " Make sure it is a git repository.", dir.getAbsolutePath()));
         }
-        this.repoRoot = git.getRepository().getDirectory().getParentFile();
     }
 
     @Override
     public void close() throws Exception {
-        git.close();
+        if (git != null) {
+            git.close();
+        }
     }
 
     @Override
     public Set<Change> diff() {
         final Set<Change> allChanges= new HashSet<>();
+        if (git == null) {
+            throw new IllegalStateException("Git is not initialized. new, changed, affected strategies needs scm to be initialized. "
```

Comment:
same here with the concrete names. But I understand what you meant - to say which strategies cannot work if the project is not git repository.
There are two ways - return null and then throw the exception from the particular strategy, or provide strategy name as a parameter and incorporate it into the message.